### PR TITLE
Document our code proposal, review, and deployment process

### LINKED ISSUE #128: Establish a basic code proposal -> review -> test -> approval -> merge -> deploy workflow
## Acceptance Criteria
- [ ] Documenting code and design review, in docs
- [ ] Documenting the QASP and merging into Docs (tracking in #217)
- [ ] Tests we want to install
- [ ] Code to cloud
- [ ] Update README with contributing instructions

This includes CI/CD, but is more than just that. Who reviews? What constitutes approval? Who merges? Small but important things.

### PR DESCRIPTION
**Pull request by greg-does-weather**
## What's wrong?

We haven't documented those things.

## How does this PR fix it? 

Documents those things. tktk

Closes #128.

## Types of changes
- [x] 🆕 New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] My change ***is*** documentation 🤡

### CODE REVIEW

File: docs/standards.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  F--yes-->H[Merge]
+```
+
+#### Branch protections
+
+PRs cannot be merged into `main` unless they satisfy our branch protection
+rules:
+
+- All PRs must be reviewed and approved by at least one other person.
+- All PRs must pass all required automated checks. The list of required checks
```
Comment (loganmcdonald-noaa) [inline docs/standards.md:99]: All #noqa must be documented or defended in the PR? There are definitely going to be false flags in tests and times we'll want to ignore things.

File: docs/standards.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  will evolved over time.
+- All commits in a PR must be cryptographically signed by their authors.
```
Comment (loganmcdonald-noaa) [inline docs/standards.md:101]: Link to documentation about how to set this up?

**Review by loganmcdonald-noaa — APPROVED** [review]
Some comments but overall LGTM

File: docs/standards.md (hunk 1 above)
Comment (greg-does-weather) [reply docs/standards.md:99]: Good catch. I had to google "noqa" but I get it now. 😆 Will update.

File: docs/standards.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+before deployhing it.
+
+> [!WARNING]  
+> Even though pushing to the `main` branch is disabled, tags applied to commits
+> on the `main` branch **_can_** be pushed. Be very careful with manually
+> tagging the repo. If your tag matches the `vX.Y` format indicating a new
+> production release, it **_will_** trigger a deployment into production.
+
+> [!NOTE]  
+> In the future, our deployment strategy should also include building provenance
```
[the remark this answers is not in the document]
Comment (mgwalker) [reply docs/standards.md:130]: Sure! I first ran into this with GitHub's npm provenance work: https://github.blog/2023-04-19-introducing-npm-package-provenance/. The idea is to be able to verify builds came from the time and place they claim to by having the build metadata signed by a trusted party and stored as a build artifact.

**Comment by greg-does-weather** [comment]
Going to hold on to this until our existing PRs have merged since they may contain unsigned commits. The signed commits branch protection rule would slow those up.

File: docs/standards.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+from that tag. We will create tags via GitHub releases rather than creating them
+manually so we can associate them with release notes, documenting what is
+changing. Using tags also allows us an opportunity to review an entire release
+candidate as a whole before deploying it.
+
+> [!WARNING]  
+> Even though pushing to the `main` branch is disabled, tags applied to commits
+> on the `main` branch **_can_** be pushed. Be very careful with manually
+> tagging the repo. If your tag matches the `vX.Y` format indicating a new
+> production release, it **_will_** trigger a deployment into production.
```
Comment (colinmurphy01) [inline docs/standards.md:136]: Is there a way to remind us of this? maybe in a PR checklist?

Comment (greg-does-weather) [reply docs/standards.md:136]: Not really. I expect manual tagging to be rare anyway, and even when we do it, I think we're almost certain to never accidentally run into this scenario. But tags are not associated with pull requests or anything else, so a reminder in a PR wouldn't really be helpful. Tags are mostly just a human-readable reference to a specific commit in the git history, so they're out of cycle from all the rest of our work.

This particular section is also just an idea of how we might trigger deploys so we've got a place in our documentation for it. We'll fill it in with real information soon and it's possible we won't use tags at all. As long as we are using GitHub Actions to run our deployments, I'd argue we should trigger deploys from releases, not tags, as we have more control and auditability over those events.

**Pull request merged**